aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -136,8 +156,17 @@ module AptosFramework::Coin {
             exists<BurnCapability<CoinType>>(account_addr),
             Errors::not_published(ENO_BURN_CAPABILITY),
         );
-        let _cap = borrow_global<BurnCapability<CoinType>>(account_addr);
 
+        let cap = borrow_global<BurnCapability<CoinType>>(account_addr);
+        burn_with_capability(coin, cap);
+    }
+
+    /// Burn `coin` with capability.
+    /// The capability `_cap` should be passed as a reference to `BurnCapability<CoinType>`.
+    public fun burn_with_capability<CoinType>(
+        coin: Coin<CoinType>,
+        _cap: &BurnCapability<CoinType>,
```

Comment:
Does passing a reference make sense? This would allow anyone to burn their own coins as they can just do borrow_global<BurnCapability<CoinType>>(owner_account_addr). I'm also addressing this in a separate PR https://github.com/aptos-labs/aptos-core/pull/921 but this should potentially be a struct object instead of reference. More on this pattern with passing Cap<T> here: https://github.com/diem/diem/tree/main/diem-move/diem-framework/core